### ISSUE TITLE
should trigger addNotificationReceivedListener when app in background

### ISSUE THREAD

**Issue by sa1dai** [expo/expo#9859]
## 🐛 Bug Report

Should trigger addNotificationReceivedListener when app in background.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

Expo CLI 3.24.2 environment info:
    System:
      OS: Linux 5.4 Ubuntu 18.04.4 LTS (Bionic Beaver)
      Shell: 5.4.2 - /usr/bin/zsh
    Binaries:
      Node: 14.4.0 - ~/.nvm/versions/node/v14.4.0/bin/node
      Yarn: 1.22.0 - /usr/bin/yarn
      npm: 6.14.5 - ~/.nvm/versions/node/v14.4.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    npmPackages:
      expo: ~38.0.8 => 38.0.9 
      react: ~16.11.0 => 16.11.0 
      react-dom: ~16.11.0 => 16.11.0 
      react-native: https://github.com/expo/react-native/archive/sdk-38.0.2.tar.gz => 0.62.2 
      react-native-web: ~0.11.7 => 0.11.7 
    npmGlobalPackages:
      expo-cli: 3.24.2

### Reproducible Demo

See steps to reproduce
 
### Steps to Reproduce

1. expo init --npm
2. replace App.js code with code from example snack: https://docs.expo.io/versions/latest/sdk/notifications/
3. add useNextNotificationsApi to app.json
4. increase notification trigger from 2 to 5 seconds (in App.js in function sendPushNotification)
5. add console.log in addNotificationReceivedListener handler to see that it fires 
5. expo start
6. scan qr code with real android or ios device
7. tap 'Press to schedule a notification'
8. immediately after tap press device home button to hide app (but not close, just hide)
9. notification received but no console message from addNotificationReceivedListener handler

### Expected Behavior vs Actual Behavior

 notification received but no console message from addNotificationReceivedListener handler in step 10

vs

 notification received and console message from addNotificationReceivedListener handler appeared in step 10

**Comment by cruzach** [comment]
`addNotificationReceivedListener` is fired when a notification is received while your app is foregrounded, not backgrounded. Unless the user interacts with the notification (e.g. by tapping on it, which will trigger the notificationResponseReceivedListener`), your app can't take action upon receiving a push notification if it's not in the foreground. The feature you're looking for is known as a "silent" push notification, and we have an open feature request for that here- https://expo.canny.io/feature-requests/p/silent-push-notifications

**Comment by sa1dai** [comment]
@cruzach 
This flow is working with LegacyNotifications, that is why I created the issue

**Comment by cruzach** [comment]
Can you share a reproducible demo for that? I tried with this snack - https://snack.expo.io/@charliecruzan/notifications9859 - and the LegacyNotifications listener does not fire when the app is backgrounded

**Comment by sa1dai** [comment]
Ok, I will make demo tomorrow

**Comment by sa1dai** [comment]
@cruzach 
I tested your [snack ](https://snack.expo.io/@charliecruzan/notifications9859).
I can receive event, but it works only on Android.
This is what I did:
1. Create blank template, then copy code from your snack to App.js
2. Start expo and scan qr code with real android device
3. Copy expo token to clipboard
4. Background app (press home button)
5. Then I send push notification via https://expo.io/notifications (I have expo token after app start)
6. I got console message Legacy notifications when app was background

Yeah, on iOS this doesn't work, but maybe this information will help with implementing silent notifications.
